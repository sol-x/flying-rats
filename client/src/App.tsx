import 'destyle.css';
import './App.css';
import React, {useState, useEffect} from 'react';
import {API_URL} from './constants';
import RatDetails from "./RatDetails";

const App: React.FC = () => {
  const [ratsNames, setRatsNames] = useState([]);
  const [selectedRat, setSelectedRat] = useState();

  useEffect(() => {
    fetch(`${API_URL}/rat-names`)
      .then(res => res.json())
      .then(response => {
        setRatsNames(response);
      })
      .catch(error => console.log(error));
  }, []);

  return (
    <div className="container">
      <div className="select" data-testid="rat-select">
        <select onChange={(event) => setSelectedRat(event.target.value)}>
          <option value="">No Rat</option>
          {
            ratsNames.map(name => <option key={name} value={name}>{name}</option>)
          }
        </select>
      </div>
      <RatDetails name={selectedRat}/>
    </div>
  );
};

export default App;
