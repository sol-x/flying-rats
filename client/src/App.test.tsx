import React from "react";
import App from "./App";
import {render, waitForElement} from "@testing-library/react";
import fetchMock from "fetch-mock";
import {API_URL} from "./constants";

beforeAll(() => {
  // @ts-ignore
  global.fetch = fetch;
});

afterAll(() => {
  fetchMock.restore();
});

it("rads select renders successfully", async () => {
  fetchMock.restore().get(`begin:${API_URL}`, ['rat 1', 'rat 2']);

  const { getByTestId, getByText } = render(<App />);
  const ratSelectNode = await waitForElement(() =>
    getByTestId('rat-select')
  );
  expect(ratSelectNode).toMatchSnapshot();
  expect(getByText('No Rat')).toBeDefined();
  expect(getByText('rat 1')).toBeDefined();
});
