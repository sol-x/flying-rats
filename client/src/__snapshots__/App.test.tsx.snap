// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rads select renders successfully 1`] = `
<div
  class="select"
  data-testid="rat-select"
>
  <select>
    <option
      value=""
    >
      No Rat
    </option>
    <option
      value="rat 1"
    >
      rat 1
    </option>
    <option
      value="rat 2"
    >
      rat 2
    </option>
  </select>
</div>
`;
